### PR TITLE
Smile 1223 template scopus s3 and unzipper

### PR DESCRIPTION
**Pull request by nraanes**
Tuned role policy for nva-scopus function and unzipper-function. Unzipper should now be (almost) PEP8 compliant.

### CODE REVIEW

**Review by anetteOlli — APPROVED** [review]

File: template.yaml [hunk 1]
```diff
@@ -321,73 +321,86 @@ Resources:
...
+      Policies:
+        - PolicyName: ScopusUnzipperBucketAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:PutObject
+                Resources:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}-${AWS::AccountId}/*'
```
Comment (axthosarouris) [inline template.yaml:354]: This is a reference to the bucket's name  (i.e.,  !GetAtt ScopuZipBucket.Name). Adding the account id is not necessary

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: template.yaml (hunk 1 above)
Comment (axthosarouris) [reply template.yaml:354]: It is best to use the parameters as little as possible and use the !Ref and !GetAtt functions using the actual AWS entities. This gives more flexibility

File: template.yaml [hunk 2]
```diff
@@ -321,73 +321,86 @@ Resources:
...
+                Action:
+                  - s3:PutObject
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}/*'
+                  - !Sub 'arn:aws:s3:::${ScopusXmlBucket}/*'
+              - Effect: Allow
+                Action:
+                  - s3:GetObject
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}/*'
```
Comment (nraanes) [inline template.yaml:360]: When I test creating stack from template I run into this error:

_Circular dependency between resources: [ScopusUnzipperFunctionUnzipEventPermission, ScopusUnzipperFunction, ScopusZipBucket, ScopusUnzipperFunctionRole]_

However, when not referencing the actual bucket in the policy statements the stack is created: `'arn:aws:s3:::${ScopusZipBucketName}-${AWS::AccountId}/*'`

Comment (nraanes) [reply template.yaml:360]: I seem to have used ${} wrong. Will look at this again tomorrow.

File: template.yaml [hunk 3]
```diff
@@ -340,7 +340,7 @@ Resources:
   ScopusBucketsAccessPolicy:
     Type: AWS::IAM::Policy
     Properties:
-      PolicyName: !Sub DefaultLambdaPermissions-${AWS::StackName}
+      PolicyName: !Sub ScopusBucketsAccessPolicy-${AWS::StackName}
```
Comment (nraanes) [inline template.yaml:343]: Bugfix: DefaultLambdaPermissions-policy was overwritten

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**